### ISSUE TITLE
False positive: Sub is required if a variable is used in a string in parameter descriptions

### ISSUE THREAD

**Issue by danieljamesscott** [aws-cloudformation/cfn-python-lint#1252]
*cfn-lint version: 0.26.0*

*Description of issue.*
Parameter descriptions fail E1029 if they contain text which looks like variable substitution:

e.g.

```yaml
  MyContentBucket:
    Description: "Bucket name for content (usually ${VPCName}-my-content), use 'none' to disable creation"
    Type: String
```

Gives an error:

    [E1029: Sub is required if a variable is used in a string] (Found an embedded parameter outside of an "Fn::Sub" at Parameters/MyContentBucket/Description)
